dali/pipeline/data/dynamic_tensor_view_test.cc

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <numeric>
+#include <utility>
+
+#include "dali/c_api.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/core/tensor_view.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/data/dynamic_tensor_view.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+using EBT = EmptyBackendTag;  // allow most of the checks to fit in one line
+
+template <typename DynamicTV, int ndim>
+void compare(const DynamicTV &dtv, const void *ptr, const TensorShape<ndim> &shape,
+             DALIDataType dtype) {
```

Comment:
I think it should be a macro, not a function - this way you'll get meaningless line info when the test fails. GTest doesn't like this kind of helpers and I'm guilty of having created ones, too.